### PR TITLE
chore!: Drop node 14 support, and removes the --experimental-wasm-flag.

### LINKED ISSUE #114: Remove --experimental-wasm-bigint flag when wasm-bigint is no longer experimental
Once this feature in Node.js is no longer experimental, we can remove this flag.

https://github.com/grain-lang/grain/blob/master/cli/bin/grain.js#L2

### PR DESCRIPTION
**Pull request by spotandjake**
Switches the engine version and github workflow to use node 16 instead of node 14, along with removes the use of the `--expermental-wasm-flag`.

closes #114.

### CODE REVIEW

**Comment by phated** [comment]
Updated title, this is a breaking change.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest]
-        node-version: ["14", "16"]
+        node-version: ["16"]
```
Comment (ospencer) [inline .github/workflows/ci.yml:13]: @phated Should we consider adding 17 here?

Comment (phated) [reply .github/workflows/ci.yml:13]: Never consider an odd node version. Odd node versions are pre-release versions and can break at any time with no semantic meaning.

**Review by peblair — APPROVED** [review]

**Review by phated — APPROVED** [review]
Thanks @spotandjake! Sorry this took so long to land

File: cli/bin/grain.js [hunk 2]
```diff
@@ -10,12 +10,6 @@ const v8 = require("v8");
  *
  * This seems to work for our needs with Node 14, but we should be cautious when updating.
```
Comment (ospencer) [inline cli/bin/grain.js:11]:
```suggestion
 * This seems to work for our needs with Node 16, but we should be cautious when updating.
```

**Review by ospencer — APPROVED** [review]
Thank you @spotandjake!!!

**Pull request merged**